src/components/create-user.html

```diff
@@ -44,31 +42,90 @@
-          <input type="text" name="firstName" id="firstName">
-        </div>
-        <div>
-          <label for="lastName">Last name:</label>
-          <input type="text" name="lastName" id="lastName">
-        </div>
-        <div>
-          <label for="phone">Phone:</label>
-          <input type="text" name="phone" id="phone">
-        </div>
-        <div>
-          <label for="email">Email:</label>
-          <input type="text" name="email" id="email">
-        </div>
-        <button type="submit">Save</button>
-      </fieldset>
-    </form>
+    <template is="dom-if" if="[[addUserSelected]]">
+        <button class="add-button" type="button" on-click="createNewUserProfile">[[changeButtonText(addUserSelected)]]</button>
+    </template>
+    <template is="dom-if" if="[[!addUserSelected]]">
+      <table>
+        <tr>
+          <td><label for="firstName">First name :</label></td>
+          <td><input type="text" name="firstName" id="firstName" value="{{user.firstName}}" on-input="updateFirstName"></td>
+        </tr>
+        <tr>
+          <td><label for="lastName">Last name :</label></td>
+          <td><input type="text" name="lastName" id="lastName" value="{{user.lastName}}" on-input="updateLastName"></td>
+        </tr>
+        <tr>
+          <td><label for="phone">Phone :</label></td>
+          <td><input type="text" name="phone" id="phone" value="{{user.phoneNumber}}" on-input="updatePhoneNumber"></td>
+        </tr>
+        <tr>
+          <td><label for="email">Email :</label></td>
+          <td><input type="text" name="email" id="email" value="{{user.email}}" on-input="updateEmail"></td>
+        </tr>
+        <tr>
+          <td><button type="button" on-click="newProfile">[[changeButtonText(addUserSelected)]]</button></td>
+        </tr>
+      </table>
+    </template>
   </template>
   <script type="module">
     class CreateUser extends Polymer.Element {
       static get is() { return 'create-user'; }
+
+      static get properties() {
+        return {
+          addUserSelected: {
+            type: Boolean,
+            value: true
+          },
+          user: {
+            type: Object,
+            value: function() {
+              return {
+                id: "0",
+                firstName: '',
+                lastName: '',
+                phoneNumber: 0,
+                email: "@banno.com" 
+              };
+            }
+          }
+        };
+      }
+
+      newProfile(event) {
+        this.dispatchEvent(new CustomEvent("newUser", {
+          bubbles: true,
+          composed: true,
+          detail: this.user
+        }));
+        this.addUserSelected = !this.addUserSelected;
+      }
+
+      createNewUserProfile(event) {
+        this.addUserSelected = !this.addUserSelected;
+      }
+
+      changeButtonText(addUserSelected) {
+        return addUserSelected ? `Create User` : `Save`;
+      }
+
+      updateFirstName(event) {
```

Comment:
If you are using the two way data binding above, are these functions even necessary?